python/lsst/daf/butler/formatters/jsonFormatter.py

```diff
@@ -78,7 +78,50 @@ def _writeFile(self, inMemoryDataset, fileDescriptor):
         with open(fileDescriptor.location.path, "w") as fd:
             if hasattr(inMemoryDataset, "_asdict"):
                 inMemoryDataset = inMemoryDataset._asdict()
-            json.dump(inMemoryDataset, fd)
+            fd.write(self._toBytes(inMemoryDataset))
+
+    def _fromBytes(self, inMemoryDataset, pytype=None):
+        """Read the bytes object as a python object.
+
+        Parameters
+        ----------
+        pickledDataset : `bytes`
+            Bytes object to unserialize.
+        pytype : `class`, optional
+            Not used by this implementation.
+
+        Returns
+        -------
+        data : `object`
+            Either data as Python object read from the pickled string, or None
+            if the string could not be read.
+        """
+        try:
+            data = json.loads(inMemoryDataset)
+        except json.JSONDecodeError:
+            data = None
+
+        return data
+
+    def _toBytes(self, inMemoryDataset):
+        """Write the in memory dataset to a bytestring.
+
+        Parameters
+        ----------
+        inMemoryDataset : `object`
+            Object to serialize
+
+        Returns
+        -------
+        data : `bytes`
+            Bytes object representing the pickled object.
+
+        Raises
+        ------
+        Exception
+            The object could not be pickled.
+        """
+        return json.dumps(inMemoryDataset)
```

Comment:
You might need to add `.encode()` to the end of this to convert the string to bytes. Consider also adding `ensure_ascii=False`.

Reply:
Both added.